src/main/scala/nl.knaw.dans.easy.deposit/Submitter.scala

```diff
@@ -34,46 +36,93 @@ import scala.util.Try
-    datasetMetadata <- depositDir.getDatasetMetadata // TODO skip recover: internal error if not catched by getDOI
-    agreementsXml <- AgreementsXml(depositDir.user, DateTime.now, datasetMetadata)
-    datasetXml <- DatasetXml(datasetMetadata)
-    msg = datasetMetadata.messageForDataManager.getOrElse("")
-    filesXml <- FilesXml(dataFilesDir)
-    // EASY-1464 3.3.8.a create empty staged bag to take a copy of the deposit
-    stageDir = (stagingBaseDir / depositDir.id.toString).createDirectories()
-    stageBag <- DansV0Bag.empty(stageDir / "bag").map(_.withEasyUserAccount(depositDir.user))
-    // EASY-1464 3.3.6 change state and copy with the rest of the deposit properties to staged dir
-    _ <- depositDir.setStateInfo(StateInfo(StateInfo.State.submitted, "Deposit is ready for processing."))
-    propsFileName = "deposit.properties"
-    _ = (dataFilesDir.parent.parent / propsFileName).copyTo(stageDir / propsFileName)
-    // EASY-1464 3.3.5.b: write files to metadata
-    _ = stageBag.addMetadataFile(msg, "message-from-depositor.txt")
-    _ <- stageBag.addMetadataFile(agreementsXml, "agreements.xml")
-    _ <- stageBag.addMetadataFile(datasetXml, "dataset.xml")
-    _ <- stageBag.addMetadataFile(filesXml, "files.xml")
-    // TODO: the next steps in a worker thread so that submit can return fast for large deposits.
-    // EASY-1464 3.3.7 checksums +  3.3.8.b copy files
-    _ <- stageBag.addPayloadFile(dataFilesDir, Paths.get("."))
-    _ <- stageBag.save() // TODO after each file to allow resume?
+  def submit(depositDir: DepositDir): Try[Unit] = {
+    val propsFileName = "deposit.properties"
+    for {
+      // TODO cache json read (and possibly rewritten) by getDOI and  getDatasetMetadata?
+      // EASY-1464 step 3.3.1 - 3.3.3
+      _ <- depositDir.getDOI(pidRequester)
+      // EASY-1464 step 3.3.4 validation
+      //   [v] mandatory fields are present and not empty (by DatasetXml(datasetMetadata) in createXMLs)
+      //   [v] DOI in json matches properties (by getDOI)
+      //   [ ] URLs are valid
+      //   [ ] ...
+      // EASY-1464 3.3.5.a: generate (with some implicit validation) content for metadata files
+      draftBag <- depositDir.getDataFiles.map(_.bag)
+      datasetMetadata <- depositDir.getDatasetMetadata // TODO skip recover: internal error if not catched by getDOI
+      agreementsXml <- AgreementsXml(depositDir.user, DateTime.now, datasetMetadata)
+      datasetXml <- DatasetXml(datasetMetadata)
+      msg = datasetMetadata.messageForDataManager.getOrElse("")
+      filesXml <- FilesXml(draftBag.data)
+      _ <- sameFiles(draftBag.payloadManifests, draftBag.baseDir / "data")
+      // TODO this is the last moment to report an invalid DDM as user error, subsequent errors are internal server errors
+      // EASY-1464 3.3.8.a create empty staged bag to take a copy of the deposit
+      stageDir = (stagingBaseDir / depositDir.id.toString).createDirectories()
```

Comment:
Perhaps we want `(stagingBaseDir / user / depositDir.id.toString)` as we have for drafts? It might make it easier to set the status to Failed or Rejected for both when problems occur once the status was changed to submitted. Especially in the (not yet implemented) copy-thread. 

Currently the state is left as submitted on later problems, but we already are getting off-topic by adding validation now that the draft bag has content in the payload manifest to build the file list.